### PR TITLE
started editing docs to add "how to get help"

### PR DESCRIPTION
**Pull request by safay**
*draft* edits for "Supportability tasks #296," how to ask for help

### CODE REVIEW

**Comment by safay** [comment]
@mr-c:
I need some clarification about what you mean by "getting help through github"

**Comment by mr-c** [comment]
@safay Creating an issue: https://github.com/ged-lab/khmer/issues/new

**Comment by safay** [comment]
@mr-c: 
Ah okay, literally reporting an issue.  Thanks for the clarification!

**Comment by safay** [comment]
- [ ] Is it mergable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage.
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

**Comment by safay** [comment]
@mr-c: How's it look? Ready to merge?

File: doc/user/getting-help.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+===============
+How to get help
+===============
+
+**First**, be sure that you:
+
+#. Read the documentation (this site)
+
+#. Work through relevant examples in the `khmer-recipes repository <https://github.com/ged-lab/khmer-recipes>`__. Find **khmer-recipes** `documentation here <http://khmer-recipes.readthedocs.org/en/latest/#>`__.
```
Comment (mr-c) [inline doc/user/getting-help.rst]: Wrap lines at 80 characters, please :-)

File: doc/user/getting-help.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Asking a good question
+----------------------
+
+#. Be very specific in your subject line.
+  - If well written, it will more likely catch the eye of someone who knows the solution.
+  - Subject lines like *"arrgh cant get it to work"* or *"NEED HELP, POSTER DUE TOMORROW!!!!!"* are not likely to inspire anyone to help.
+  - Example of a good subject line: *"Error message from normalize-by-median 'Hash writing file access failure:: Bad address'"*
+
+2. Include your OS version, Python version, and khmer version.
```
Comment (mr-c) [inline doc/user/getting-help.rst]: Specific commands to run will produce better results here

File: doc/user/getting-help.rst (hunk 1 above)
Comment (safay) [inline doc/user/getting-help.rst]: @mr-c: I'm not sure how to put line breaks without screwing up the restructured text formatting; can you advise?

**Comment by safay** [comment]
re-wrote google search box from scratch; this is my code

**Comment by safay** [comment]
@mr-c: when you have some time please have a look again; I think this is ready

File: ChangeLog [hunk 3]
```diff
@@ -1,3 +1,7 @@
+2015-04-14  Scott Fay  <scott.a.fay@gmail.com>
+
+   * Added how-to-help to user docs
```
Comment (mr-c) [inline ChangeLog]: Please list each file modified :-)

File: README.rst [hunk 4]
```diff
@@ -43,6 +43,9 @@ them, please visit the following URLs:
 We chat at https://gitter.im/ged-lab/khmer and the maintainers can be
 contacted at khmer-project@idyll.org.
 
+Detailed guide to getting help with khmer:
```
Comment (mr-c) [inline README.rst]: Can you rephrase as a complete sentence matching the tone above?

File: doc/user/getting-help.rst [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+How to get help
+===============
+
+First, be sure that you:
+
+#. Read the documentation (this site)
+
+#. Work through relevant examples in the `khmer-recipes repository
+   <https://github.com/ged-lab/khmer-recipes>`__. Find **khmer-recipes**
+   `documentation here <http://khmer-recipes.readthedocs.org/en/latest/#>`__.
```
Comment (mr-c) [inline doc/user/getting-help.rst]: Lets not include this one

File: doc/user/getting-help.rst [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  - If well written, it will more likely catch the eye of someone who knows
+    the solution.
+  - Subject lines like *"arrgh cant get it to work"* or *"NEED HELP,
+    POSTER DUE TOMORROW!!!!!"* are not likely to inspire anyone to help.
+  - Example of a good subject line: *"Error message from normalize-by-median
+    'Hash writing file access failure:: Bad address'"*
+
+2. Include your:
+
+ a. OS version(Mac OS X or Linux):  ``uname -mrs``
```
Comment (mr-c) [inline doc/user/getting-help.rst]: space before parens

File: doc/user/getting-help.rst [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+#. Be very specific in your subject line.
+  - If well written, it will more likely catch the eye of someone who knows
+    the solution.
+  - Subject lines like *"arrgh cant get it to work"* or *"NEED HELP,
+    POSTER DUE TOMORROW!!!!!"* are not likely to inspire anyone to help.
+  - Example of a good subject line: *"Error message from normalize-by-median
+    'Hash writing file access failure:: Bad address'"*
+
+2. Include your:
```
Comment (mr-c) [inline doc/user/getting-help.rst]: `#.` instead of `2.`

File: doc/user/getting-help.rst [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+Mailing List
+------------
+
+The primary way to get help is through the khmer discussion list:
+http://lists.idyll.org/listinfo/khmer
+
+When you ask a question, it is important that you ask in a way that ensures
+you get the best answer possible and minimizes back-and-forth with the good
+people trying to help.
```
Comment (mr-c) [inline doc/user/getting-help.rst]: Can you drop this para as well?

File: doc/user/getting-help.rst [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+Asking a good question
+----------------------
+
+#. Be very specific in your subject line.
```
Comment (mr-c) [inline doc/user/getting-help.rst]: Given the low volume of requests we receive lets not add extra requirements for asking a question at this time.

File: doc/user/getting-help.rst [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  - Example of a good subject line: *"Error message from normalize-by-median
+    'Hash writing file access failure:: Bad address'"*
+
+2. Include your:
+
+ a. OS version(Mac OS X or Linux):  ``uname -mrs``
+
+ b. Python version:  ``python --version``
+
+ c. and khmer version:  ``pip freeze | grep khmer``
```
Comment (mr-c) [inline doc/user/getting-help.rst]: FYI: This will be provided if they include all the output from one of our scripts.

File: doc/user/getting-help.rst [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+#. Copy-and-paste the exact command that is causing the problem.  Include the
+   steps you performed leading up to the issue.
+
+#. Include the complete error message; if it is large, include a link to a
+   file.
+
+GitHub
+------
+
+If you think you've found a legitimate bug or other problem with
```
Comment (mr-c) [inline doc/user/getting-help.rst]: I would say: "You are also welcome to report an issue you are having using GitHub:"
